src/test/kotlin/no/nb/mlt/wls/product/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,212 @@
+
+    @Test
+    @WithMockUser
+    fun `createProduct where SynQ says it's a duplicate returns OK`() {
+        every {
+            synqProductService.createProduct(any())
+        } returns ResponseEntity.ok().build()
+
+        webTestClient
+            .mutateWith(csrf())
+            .post()
+            .accept(MediaType.APPLICATION_JSON)
+            .bodyValue(testProductPayload)
+            .exchange()
+            .expectStatus().isOk
+            .expectBody().isEmpty
+    }
+
+    @Test
+    @WithMockUser
+    fun `createProduct handles SynQ error`() {
+        every {
+            synqProductService.createProduct(any())
+        }.throws(
+            ServerErrorException(
+                "Failed to create product in SynQ, the storage system responded with " +
+                    "error code: '1002' and error text: 'Unknown product category TEST.'",
+                HttpClientErrorException(HttpStatus.NOT_FOUND, "Not found")
+            )
+        )
+
+        webTestClient
+            .mutateWith(csrf())
+            .post()
+            .accept(MediaType.APPLICATION_JSON)
+            .bodyValue(testProductPayload)
+            .exchange()
+            .expectStatus().is5xxServerError
+    }
+
+// /////////////////////////////////////////////////////////////////////////////
+// //////////////////////////////// Test Help //////////////////////////////////
+// /////////////////////////////////////////////////////////////////////////////
+
+    // Will be used in most tests
+    private val testProductPayload =
+        ApiProductPayload(
+            hostId = "mlt-420",
+            hostName = HostName.AXIELL,
+            description = "Ringenes Herre samling",
+            productCategory = "BOOK",
+            preferredEnvironment = NONE,
+            packaging = Packaging.BOX,
+            owner = Owner.NB,
+            location = "SYNQ_WAREHOUSE",
+            quantity = 1.0
+        )
+
+    // Will exist in the database
+    private val duplicateProductPayload =
+        ApiProductPayload(
+            hostId = "product-12346",
+            hostName = HostName.AXIELL,
+            description = "Tyv etter loven",
+            productCategory = "BOOK",
+            preferredEnvironment = NONE,
+            packaging = Packaging.NONE,
+            owner = Owner.NB,
+            location = "SYNQ_WAREHOUSE",
+            quantity = 1.0
+        )
+
+    fun populateDb() {
+        // Make sure we start with clean DB instance for each test
+        repository.deleteAll()
+        repository.save(
+            Product(
+                hostId = "product-12346",
+                hostName = HostName.AXIELL,
+                description = "Tyv etter loven",
+                productCategory = "BOOK",
+                preferredEnvironment = NONE,
+                packaging = Packaging.NONE,
+                owner = Owner.NB,
+                location = "SYNQ_WAREHOUSE",
+                quantity = 1.0
+            )
+        )
```

Comment:
I swear there was some good reason for why I made them separate, since for a while I did had it as `duplicateProductPayload.toProduct()` but then changed it back. But now it's gone in the dank caverns of my mind.